compiler/optimizer/OMROptimizationManager.cpp

```diff
@@ -49,7 +49,7 @@ OMR::OptimizationManager::OptimizationManager(TR::Optimizer *o, OptimizationFact
         _numPassesCompleted(0),
         _optData(NULL),
         _optPolicy(NULL),
-        _enabled(false),
+        _enabled(!o->comp()->isDisabled(optNum)),
```

Comment:
do you always have an opt index in hand to be able to do the regex test? .... I don't think Opt A can ask "is Opt B disabled" with the regex unless A knows what B's opt index *will* be.

This suggests to me that asking the isDisabled() question via the regex is a more "timely" kind of query (you have to ask it "in the moment", as it were) rather than a timeless notion implied by the other query. Maybe (and I'll be impressed if someone actually put this much thought into it) it's called "isDisabled()" because returning true means the opt won't run, but returning false doesn't necessarily mean it will run (because it might fail the regex check). Normally I don't like "negative" queries because I hate to see code like !isDisable() but, in this case, it might be the most correct question to ask...